modules/peskyNeighbor.ts

```diff
@@ -0,0 +1,168 @@
+import { WebSocket } from "ws";
+import fs from "node:fs/promises";
+import { createReadStream } from "node:fs";
+import { SLACK_PRIVATE_CHANNEL_ID, SLACK_TEAM_ID, SLACK_GATEWAY_SERVER } from "../config";
+import { SLACK_USER_TOKEN, SLACK_USER_COOKIE } from "../auth";
+
+const HEADERS = {
+	cookie: `d=${encodeURIComponent(SLACK_USER_COOKIE)}`,
+};
+const SLEEP = 10_000;
+
+function createSlackSocket() {
+	const websocketParams = {
+		token: SLACK_USER_TOKEN,
+		sync_desync: "1",
+		slack_client: "desktop",
+		start_args:
+			"?agent=client&org_wide_aware=true&agent_version=1742590350&eac_cache_ts=true&cache_ts=0&name_tagging=true&only_self_subteams=true&connect_only=true&ms_latest=true",
+		no_query_on_subscribe: "1",
+		flannel: "3",
+		lazy_channels: "1",
+		gateway_server: SLACK_GATEWAY_SERVER,
+		batch_presence_aware: "1",
+	};
+	const socket = new WebSocket(
+		`wss://wss-primary.slack.com/?${new URLSearchParams(websocketParams)}`,
+		{
+			origin: "https://app.slack.com",
+			headers: HEADERS,
+		}
+	);
+
+	return socket;
+}
+
+async function fetchSlackEndpoint(
+	command: string,
+	params: Record<string, string | readonly string[]>
+) {
+	let body = new FormData();
+	body.append("token", SLACK_USER_TOKEN);
+	let resp = await fetch(`https://slack.com/api/${command}?${new URLSearchParams(params)}`, {
+		headers: HEADERS,
+		body,
+		method: "POST",
+	}).then((resp) => resp.json());
+
+	if (!resp.ok) {
+		throw new Error(
+			`Fetching ${command} with ${JSON.stringify(params)} resulted in error: ${resp.error}`
+		);
+	}
+
+	return resp;
+}
+
+function listenOnce(socket: WebSocket, type: string) {
+	return new Promise((resolve, reject) => {
+		function callback(event) {
+			let data = JSON.parse(event.toString());
+			if (data.type === type) {
+				resolve(data);
+				socket.removeListener("message", callback);
```

Comment:
Isn't this just `socket.once(callback)`?

Reply:
No, because it waits for the first event of a certain type